lib/Serialization/ModuleFileSharedCore.cpp

```diff
@@ -364,7 +364,7 @@ static ValidationInfo validateControlBlock(
         version::isCurrentCompilerTagged();
 
       StringRef moduleRevision = blobData;
-      if (isCompilerTagged) {
+      if (isCompilerTagged && requiresRevisionMatch) {
```

Comment:
seems you could also do an early-`break` near the top of this `case`.

Reply:
I guess I should also honor the force case...

Reply:
make sense